Warning with newer GCC versions in matrix_free/fe_evaluation.h

```
[226/939] Building CXX object examples/CMakeFiles/step-76.release.dir/step-76/step-76.cc.o
In file included from /home/bangerth/p/deal.II/1/dealii/examples/step-76/step-76.cc:46:
In member function ‘void dealii::FEEvaluationBase<dim, n_components_, Number, <anonymous>, <template-parameter-1-5> >::read_write_operation(const VectorOperation&, const std::array<VectorType*, n_components_>&, const std::array<const std::vector<dealii::ArrayView<const typename VectorType::value_type> >*, n_components_>&, const std::bitset<VectorizedArrayType::size()>&, bool) const [with VectorType = dealii::LinearAlgebra::distributed::Vector<double>; VectorOperation = dealii::internal::VectorDistributorLocalToGlobal<double, dealii::VectorizedArray<double, 2> >; int dim = 2; int n_components_ = 4; Number = double; bool is_face = false; VectorizedArrayType = dealii::VectorizedArray<double, 2>]’,
    inlined from ‘void dealii::FEEvaluationBase<dim, n_components_, Number, <anonymous>, <template-parameter-1-5> >::distribute_local_to_global(VectorType&, unsigned int, const std::bitset<VectorizedArrayType::size()>&) const [with VectorType = dealii::LinearAlgebra::distributed::Vector<double>; int dim = 2; int n_components_ = 4; Number = double; bool is_face = false; VectorizedArrayType = dealii::VectorizedArray<double, 2>]’ at /home/bangerth/p/deal.II/1/dealii/include/deal.II/matrix_free/fe_evaluation.h:4598:23,
    inlined from ‘Euler_DG::EulerOperator<2, 5, 7>::perform_stage(unsigned int, Euler_DG::Number, Euler_DG::Number, Euler_DG::Number, const dealii::LinearAlgebra::distributed::Vector<double>&, dealii::LinearAlgebra::distributed::Vector<double>&, dealii::LinearAlgebra::distributed::Vector<double>&) const::<lambda(const auto:56&, auto:57&, const auto:58&, auto:59)> [with auto:56 = dealii::MatrixFree<2, double, dealii::VectorizedArray<double, 2> >; auto:57 = dealii::LinearAlgebra::distributed::Vector<double>; auto:58 = dealii::LinearAlgebra::distributed::Vector<double>; auto:59 = std::pair<unsigned int, unsigned int>]’ at /home/bangerth/p/deal.II/1/dealii/examples/step-76/step-76.cc:939:47:
/home/bangerth/p/deal.II/1/dealii/include/deal.II/matrix_free/fe_evaluation.h:3744:24: warning: ‘void* __builtin_memset(void*, int, long unsigned int)’ offset [24, 2040] is out of the bounds [0, 16] of object ‘dof_indices’ with type ‘const unsigned int* [2]’ [-Warray-bounds]
 3744 |         dof_indices[v] = nullptr;
      |         ~~~~~~~~~~~~~~~^~~~~~~~~
/home/bangerth/p/deal.II/1/dealii/include/deal.II/matrix_free/fe_evaluation.h: In lambda function:
/home/bangerth/p/deal.II/1/dealii/include/deal.II/matrix_free/fe_evaluation.h:3669:23: note: ‘dof_indices’ declared here
 3669 |   const unsigned int *dof_indices[n_lanes];
      |                       ^~~~~~~~~~~
```

Also:
```
/home/bangerth/p/deal.II/1/dealii/include/deal.II/matrix_free/fe_evaluation.h:3710:24: warning: ‘void* __builtin_memset(void*, int, long unsigned int)’ offset [24, 2040] is out of the bounds [0, 16] of object ‘dof_indices’ with type ‘const unsigned int* [2]’ [-Warray-bounds]
 3710 |         dof_indices[v] = nullptr;
      |         ~~~~~~~~~~~~~~~^~~~~~~~~
```

--- a/include/deal.II/matrix_free/fe_evaluation.h
+++ b/include/deal.II/matrix_free/fe_evaluation.h
@@ -3666,7 +3666,10 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
       return;
     }
 
-  const unsigned int *dof_indices[n_lanes];
+  // Allocate pointers, then initialize all of them to nullptrs and
+  // below overwrite the ones we actually use:
+  std::array<const unsigned int *, n_lanes> dof_indices;
+  dof_indices.fill(nullptr);
 
   // Assign the appropriate cell ids for face/cell case and get the pointers
   // to the dof indices of the cells on all lanes
@@ -3706,8 +3709,6 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
           dof_indices[v] =
             this->dof_info->dof_indices.data() + my_index_start[0].first;
         }
-      for (unsigned int v = n_vectorization_actual; v < n_lanes; ++v)
-        dof_indices[v] = nullptr;
     }
   else
     {
@@ -3740,8 +3741,6 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
           dof_indices[v] =
             this->dof_info->dof_indices.data() + my_index_start[0].first;
         }
-      for (unsigned int v = n_vectorization_actual; v < n_lanes; ++v)
-        dof_indices[v] = nullptr;
     }
 
   // Case where we have no constraints throughout the whole cell: Can go

